--- a/src/plugins/networkinformation/networklistmanager/qnetworklistmanagerevents.cpp
+++ b/src/plugins/networkinformation/networklistmanager/qnetworklistmanagerevents.cpp
@@ -171,11 +171,15 @@
             const auto scopedVariantClear = qScopeGuard([&variant]() { VariantClear(&variant); });
 
             const wchar_t *versions[] = { L"NA_InternetConnectivityV6", L"NA_InternetConnectivityV4" };
+#if defined(_MSC_VER)
+            const auto webHijackFlag = NLM_INTERNET_CONNECTIVITY_WEBHIJACK;
+#else
+            const auto webHijackFlag = 0x4000;
+#endif
             for (const auto version : versions) {
                 hr = propertyBag->Read(version, &variant, nullptr);
                 if (SUCCEEDED(hr)
-                    && (V_UINT(&variant) & NLM_INTERNET_CONNECTIVITY_WEBHIJACK)
-                            == NLM_INTERNET_CONNECTIVITY_WEBHIJACK) {
+                    && (V_UINT(&variant) & webHijackFlag) == webHijackFlag) {
                     return true;
                 }
             }
